ecosystem/typescript/sdk/src/plugins/ans_client.ts

```diff
@@ -0,0 +1,166 @@
+    this.provider = provider;
+    if (!ansContractsMap[this.provider.network] && !contractAddress) {
+      throw new Error("Please provide a valid contract address");
+    }
+    this.contractAddress = ansContractsMap[this.provider.network] ?? contractAddress;
+  }
+
+  /**
+   * Returns the primary name for the given account address
+   * @param address An account address
+   * @returns Account's primary name | null
+   */
+  async getPrimaryNamebyAddress(address: string): Promise<string | null> {
+    const ansResource: Gen.MoveResource = await this.provider.getAccountResource(
+      this.contractAddress,
+      `${this.contractAddress}::domains::ReverseLookupRegistryV1`,
+    );
+    const data = ansResource.data as ReverseLookupRegistryV1;
+    const { handle } = data.registry;
+    const domainsTableItemRequest = {
+      key_type: "address",
+      value_type: `${this.contractAddress}::domains::NameRecordKeyV1`,
+      key: address,
+    };
+    try {
+      const item = await this.provider.getTableItem(handle, domainsTableItemRequest);
+      return item.subdomain_name.vec[0] ? `${item.subdomain_name.vec[0]}.${item.domain_name}` : item.domain_name;
+    } catch (error: any) {
+      // if item not found, response is 404 error - meaning item not found
+      return null;
+    }
+  }
+
+  /**
+   * Returns the target account address for the given name
+   * @param name ANS name
+   * @returns Account address | null
+   */
+  async getAddressByName(name: string): Promise<string | null> {
+    const { domain, subdomain } = name.match(namePattern)?.groups ?? {};
+    if (!domain) return null;
+    if (subdomain) return this.getAddressBySubdomainName(domain, subdomain);
+    return this.getAddressByDomainName(domain);
+  }
+
+  /**
+   * Returns the account address for the given domain name
+   * @param domain domain name
+   * @example
+   * if name is `aptos.apt`
+   * domain = aptos
+   *
+   * @returns account address | null
+   */
+  private async getAddressByDomainName(domain: string) {
+    if (domain.match(nameComponentPattern) === null) return null;
+    const ansResource: { type: Gen.MoveStructTag; data: any } = await this.provider.getAccountResource(
+      this.contractAddress,
+      `${this.contractAddress}::domains::NameRegistryV1`,
+    );
+    const data = ansResource.data as NameRegistryV1;
+    const { handle } = data.registry;
+    const domainsTableItemRequest = {
+      key_type: `${this.contractAddress}::domains::NameRecordKeyV1`,
+      value_type: `${this.contractAddress}::domains::NameRecordV1`,
+      key: {
+        subdomain_name: { vec: [] },
+        domain_name: domain,
+      },
+    };
+
+    try {
+      const item = await this.provider.getTableItem(handle, domainsTableItemRequest);
+      return item.target_address.vec[0];
+    } catch (error: any) {
+      // if item not found, response is 404 error - meaning item not found
+      return null;
+    }
```

Comment:
same as above, I think consuming any exception is dangerous

Reply:
agreed. For now, I think we can return null if error code is 404, otherwise we throw an Error and I'll create a backlog ticket to introduce Logging system in the SDK.